packages/terra-date-time-picker/src/DateTimePicker.jsx

```diff
@@ -346,6 +350,10 @@ class DateTimePicker extends React.Component {
     this.setState({ isTimeClarificationOpen: false });
   }
 
+  convertDateTimeStringToMomentObject(date, time) {
+    return DateTimeUtils.updateTime(DateUtil.createSafeDate(DateUtil.convertToISO8601(date, this.state.dateFormat)), time);
```

Comment:
The previous check just did createSafeDate by itself without the need for updateTime. How come that's needed here?

Reply:
The string passed to the previous check had time along with the date in iSO format. so we had to only convert the string to moment object using createsafedate.  in our case we get only the date string from date control which i am converting into moment object using createsafedate and later appending time value retrieved from time control to the moment object using updatetime.

Here the MIN and MAX datetime will have time value along with the date. hence we have to use both date and time value entered to check whether it falls within the range.

Reply:
I'm still not seeing it though. maxDateTime and minDateTimes are already in ISO8601, and convertToISO8601 is converting the current date to that format, so I would definitely expect createSafeDate for both. However, since neither the maxDateTime or minDateTime need to use updateTime, I would expect the same to occur with this function. What happens if you leave of updateTime?

Reply:
This new method convertDateTimeStringToMomentObject0 is only called for the dates entered by the user in handleTimeChange event. for MIN and MAX dates this method is not called.